### PR TITLE
0.38 admin panel new components

### PR DESCRIPTION
**Pull request by kdoh**
Wanted to kick off some improvements to visuals in the admin panel by first adding some components to help standardize some of our page headings.  This adds some basic layout components for a page header and adds a couple examples of how to use them to achieve a few of the patterns we'll be adding in a follow up.

### CODE REVIEW

File: frontend/src/metabase/admin/components/Page.info.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+export const examples = {
+  "": (
+    <PageHeader>
+      <Subhead>Database listing page</Subhead>
+    </PageHeader>
+  ),
+  withAction: (
+    <PageHeader>
+      <PageTools>
```
Comment (kdoh) [inline frontend/src/metabase/admin/components/Page.info.js:20]: I don't love this name but I couldn't think of anything better at the time for a section that's supposed to act as a wrapper in the case you have actions, or actions and tabs.

**Comment by dpsutton** [comment]
Love seeing the jsx and examples together in `_internal`.

One thought for the pagetabs and pagetools versions. Each captures some styling that will help keep the look consistent, but it seems like each also makes no attempt to capture some other mechanics that might be reusable. I'm thinking of the `radio` pattern and perhaps which panel to display. Or for the page tools, taking in some action props and nesting them inside the page actions.

Not sure how common this kind of props are. In previous codebases (in cljs) i would expect to pass maps that included the panels and the the labels, and the higher order component would wire up the radio buttons and display the right render function depending on the radio selection.

**Comment by kdoh** [comment]
@dpsutton Thanks for taking a look. The main thing I wanted to try with this was to break out of a pattern we've had of having overly abstracted layouts where you pass "title" and "links" as props and it assumes a bunch of things and gives you a fairly rigid result.

I've found that inevitably you need to reach into those and tweak something so instead I thought it'd make sense to not really try and create any sort of higher order component and just document the proper composition of things to achieve what we currently view as the common layout variants. For things like the tabs or actions, any page using this pattern can just figure out what it needs on site, and this just helps you figure out where to put those elements. 

So in a sense, this isn't really a component I'm documenting, I'm more putting together recipes for how to combine some pre-styled wrappers. It's a bit less DRY in the classic sense, but I think it gives us more flexibility in the long run and is hopefully easier to understand and adapt.

**Comment by dpsutton** [comment]
gotcha. and i've seen that kinda stuff happen as well. I like the thought being put in and the direction.

**Pull request merged**